datarepo-clienttests/src/main/java/testscripts/SoftDeleteDataset.java

```diff
@@ -0,0 +1,215 @@
+    BulkLoadArrayResultModel bulkLoadArrayResultModel =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestFileJobResponse, BulkLoadArrayResultModel.class);
+    String fileId = bulkLoadArrayResultModel.getLoadFileResults().get(0).getFileId();
+
+    // ingest the tabular data from the JSON file we just generated
+    // generate a JSON file with the fileref
+    String jsonLine =
+        "{\"VCF_File_Name\":\"name1\", \"Description\":\"description1\", \"VCF_File_Ref\":\""
+            + fileId
+            + "\"}\n";
+    byte[] fileRefBytes = jsonLine.getBytes(StandardCharsets.UTF_8);
+    String jsonFileName = FileUtils.randomizeName("this-better-pass") + ".json";
+    String dirInCloud = "scratch/softDel";
+    String fileRefName = dirInCloud + "/" + jsonFileName;
+    String gsPath = FileUtils.createGsPath(fileRefBytes, fileRefName, testConfigGetIngestbucket);
+
+    IngestRequestModel ingestRequest =
+        new IngestRequestModel()
+            .format(IngestRequestModel.FormatEnum.JSON)
+            .table("vcf_file")
+            .path(gsPath);
+    JobModel ingestTabularDataJobResponse =
+        repositoryApi.ingestDataset(datasetSummaryModel.getId(), ingestRequest);
+
+    ingestTabularDataJobResponse =
+        DataRepoUtils.waitForJobToFinish(repositoryApi, ingestTabularDataJobResponse);
+    IngestResponseModel ingestResponse =
+        DataRepoUtils.expectJobSuccess(
+            repositoryApi, ingestTabularDataJobResponse, IngestResponseModel.class);
+    logger.info("Successfully loaded data into dataset: {}", ingestResponse.getDataset());
+
+    String datasetId = datasetSummaryModel.getId(); // TODO should these be in utils?
+    DatasetModel datasetModel = repositoryApi.retrieveDataset(datasetId);
+    String dataProject = datasetModel.getDataProject();
+    String tableName = datasetModel.getSchema().getTables().get(0).getName();
+
+    // get row ids for table  TODO combine SQL query w queryForFileRefs in DRSLookup
+    String tableRef =
+        String.format(
+            "`%s.%s.%s`",
+            dataProject,
+            "datarepo_" + datasetModel.getName(), // TODO const ni BQ utils
+            tableName);
+    String sqlQuery =
+        String.format(
+            "SELECT %s FROM %s LIMIT %s",
+            "datarepo_row_id", tableRef, 1L); // TODO what is the limit?
+    TableResult result = BigQueryUtils.queryBigQuery(dataProject, sqlQuery);
+    List<String> rowIds =
+        StreamSupport.stream(result.getValues().spliterator(), false)
+            .map(fieldValues -> fieldValues.get(0).getStringValue())
+            .collect(Collectors.toList());
+
+    logger.info("Successfully retrieved row ids for table: {}", tableName);
+
+    // write to GCS
+    String csvFileName = FileUtils.randomizeName("this-too-better-pass") + ".csv";
+    String csvFileRefName = dirInCloud + "/" + csvFileName;
+    String gcsPath = FileUtils.createGcsPath(rowIds, csvFileRefName, testConfigGetIngestbucket);
```

Comment:
For my education: Why do you have to do this manual write to GCS? Per our conversation yesterday, wouldn't the bulk file load put the file in gcs?

Reply:
I think this is the file that contains the rows to soft delete, not file data. So we have to put that file in GCS first before pointing Data Repo at it.